### ISSUE TITLE
Error: Failed to execute 'setStart' on 'Range' [Slate-React crashes]

### ISSUE THREAD

**Issue by alfirin** [Foundry376/Mailspring#921]
Hey guys I was editing an email and I got this error :
I can't see any emails anymore even after restarting.

Ubuntu 16.04
Mailspring version : 1.2.1

```
Error: Failed to execute 'setStart' on 'Range': The offset 4294967294 is larger than the node's length (1).
    at findDOMRange (/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/utils/find-dom-range.js:37:5)
    at Content._this.updateSelection (/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/components/content.js:154:46)
    at Content._this.componentDidUpdate (/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/components/content.js:121:13)
    at commitLifeCycles (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:149:308)
    at b (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:158:229)
    at m (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:169:217)
    at w (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:168:413)
    at batchedUpdates (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:171:357)
    at tc (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:54:269)
    at vd (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:71:249)
```

**Comment by liaralabs** [comment]
Seeing this exact same error on Windows 10 (v 1.2.2); however, a reboot of Mailspring fixes the issue for me

**Comment by alfirin** [comment]
I restarted the app and it dosn't work either.

**Comment by joho** [comment]
![image](https://user-images.githubusercontent.com/4092/42613646-621cddb2-85e5-11e8-9b26-60c18415bbe3.png)

Same here.
Mailspring version: 1.2.2-96fb3a99
Fedora 28

**Comment by gouthamve** [comment]
+1 this still exists on 1.4.2 on Mac.

```
Error: Failed to execute 'setStart' on 'Range': The offset 4294967294 is larger than the node's length (1).
    at findDOMRange (/Applications/Mailspring.app/Contents/Resources/app.asar/node_modules/slate-react/lib/utils/find-dom-range.js:37:5)
    at Content._this.updateSelection (/Applications/Mailspring.app/Contents/Resources/app.asar/node_modules/slate-react/lib/components/content.js:154:46)
    at Content._this.componentDidMount (/Applications/Mailspring.app/Contents/Resources/app.asar/node_modules/slate-react/lib/components/content.js:113:13)
    at commitLifeCycles (/Applications/Mailspring.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:149:193)
    at b (/Applications/Mailspring.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:158:229)
    at m (/Applications/Mailspring.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:169:217)
    at w (/Applications/Mailspring.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:168:413)
    at z (/Applications/Mailspring.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:167:32)
    at u (/Applications/Mailspring.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:165:441)
    at Object.enqueueSetState (/Applications/Mailspring.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:108:299)
    at _class.A.setState (/Applications/Mailspring.app/Contents/Resources/app.asar/node_modules/react/cjs/react.production.min.js:12:357)
    at _draftStore2.default.sessionForClientId.then.session (/src/decorators/inflates-draft-client-id.jsx:66:14)
```

This happens when I try to write a reply.

**Comment by M-Pixel** [comment]
```
Error: Failed to execute 'setStart' on 'Range': The offset 4294967295 is larger than the node's length (45).
    at findDOMRange (C:\Users\max\AppData\Local\Mailspring\app-1.5.2\resources\app.asar\node_modules\slate-react\lib\utils\find-dom-range.js:37:5)
    at Content._this.updateSelection (C:\Users\max\AppData\Local\Mailspring\app-1.5.2\resources\app.asar\node_modules\slate-react\lib\components\content.js:154:46)
    at Content._this.componentDidUpdate (C:\Users\max\AppData\Local\Mailspring\app-1.5.2\resources\app.asar\node_modules\slate-react\lib\components\content.js:121:13)
    at Hi (C:\Users\max\AppData\Local\Mailspring\app-1.5.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:227:383)
    at Fi (C:\Users\max\AppData\Local\Mailspring\app-1.5.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:218:212)
    at Gi (C:\Users\max\AppData\Local\Mailspring\app-1.5.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:217:174)
    at Ki (C:\Users\max\AppData\Local\Mailspring\app-1.5.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:232:135)
    at de (C:\Users\max\AppData\Local\Mailspring\app-1.5.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:85:207)
```

Occurred while using the up arrow to select text in a reply.

**Comment by geonnave** [comment]
Same error on Ubuntu 18.04, Mailspring 1.5.3-420ce003, while editing a reply.

```
Error: Failed to execute 'setStart' on 'Range': The offset 4294967294 is larger than the node's length (1).
    at findDOMRange (/snap/mailspring/277/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/utils/find-dom-range.js:37:5)
    at Content._this.updateSelection (/snap/mailspring/277/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/components/content.js:154:46)
    at Content._this.componentDidUpdate (/snap/mailspring/277/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/components/content.js:121:13)
    at Hi (/snap/mailspring/277/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:227:383)
    at Fi (/snap/mailspring/277/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:218:212)
    at Gi (/snap/mailspring/277/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:217:174)
    at Ki (/snap/mailspring/277/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:232:135)
    at de (/snap/mailspring/277/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:85:207)
```

**Comment by szbence89** [comment]
same on ubuntu 18.10 with Mailspring 1.5.3-420ce003

**Comment by mihir-amrelia** [comment]
Faced similar issue. Toggling 'Reading Pane' ON->OFF->ON helped.

**Comment by smashingx1** [comment]
seems like this bug has been around for a while and it has been persistent through different versions of the product. I got this error on Mailspring 1.5.7. Can we fix this issue please? Just as a side note, I tried what mihir-amrelia suggested but it didn't work for me, I'm still seeing the same error.

**Comment by CodeMouse92** [comment]
It appears that this issue has been resolved, so we are closing it. **If the problem occurs again, please create a new thread in the appropriate category on our [Discourse](https://community.getmailspring.com).**

Resolved here: 21bc2ef398e403ab064f7046240b2fcff455a5e2

We are in the process of migrating issues to Discourse, which can better facilitate discussion and discovery, and so GitHub Issues can focus on issues that are confirmed and slated for resolution in the near term. [Learn more about the changes here.](https://community.getmailspring.com/t/mailspring-community-transition/38)

We hope to see you there soon!

-The Mailspring Team
